File: packages/client-search/src/types/SynonymType.ts
export const SynonymEnum: { readonly [key: string]: SynonymType } = {
  Synonym: 'synonym',
  OneWaySynonym: 'oneWaySynonym',
  AltCorrection1: 'altCorrection1',
  AltCorrection2: 'altCorrection2',
  Placeholder: 'placeholder',
};

export type SynonymType =
  | 'synonym'
  | 'oneWaySynonym'
  | 'altCorrection1'
  | 'altCorrection2'
  | 'placeholder';

File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/SynonymType.java
package com.algolia.model;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Type of the synonym object. */
@JsonAdapter(SynonymType.Adapter.class)
public enum SynonymType {
  SYNONYM("synonym"),

  ONEWAYSYNONYM("onewaysynonym"),

  ALTCORRECTION1("altcorrection1"),

  ALTCORRECTION2("altcorrection2"),

  PLACEHOLDER("placeholder");

  private String value;

  SynonymType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static SynonymType fromValue(String value) {
    for (SynonymType b : SynonymType.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<SynonymType> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final SynonymType enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public SynonymType read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return SynonymType.fromValue(value);
    }
  }
}
